### PR TITLE
feat: Allow gprogram without exposed services

### LINKED ISSUE #202: macros: Code generated by the 'gprogram' macro leads to compilation error if there are no exposed services
### Problem

Code generated by the 'gprogram' macro leads to compilation error if there are no exposed services discovered, i.e. there are no **public** methods returning constructed service.

### Steps

```
struct MyProgram(());

[gprogram]
impl MyProgram {
    pub new() -> Self {
        Self(())
    }
}
```
leads to the error
```
error[E0277]: the trait bound `(): gstd::AsRef<[u8]>` is not satisfied
     --> /home/xxxx/sources/eeee/sails/examples/battleship/app/src/lib.rs:10:1
      |
  10  | #[gprogram]
      | ^^^^^^^^^^^ the trait `gstd::AsRef<[u8]>` is not implemented for `()`
      |
  note: required by a bound in `reply_bytes`
     --> /home/xxxx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gstd-1.3.0/src/msg/basic.rs:392:34
      |
  392 | pub fn reply_bytes(payload: impl AsRef<[u8]>, value: u128) -> Result<MessageId> {
      |                                  ^^^^^^^^^^^ required by this bound in `reply_bytes`
      = note: this error originates in the attribute macro `gprogram` (in Nightly builds, run with -Z macro-backtrace for more info)
```

### Possible Solution

There are 2 possible solutions:
1. Forbid programs, i.e. report a reasonable error, which do not expose any service
2. The above code should not lead to a compilation error

### Notes

_No response_

### Relevant Log Output

<details><summary>Click to expand/collapse</summary>
<p>

```
<Paste your log here>
```

</p>
</details>

### PR DESCRIPTION
**Pull request by holykol**
Resolves #202

### CODE REVIEW

File: examples/empty/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "conditional-service"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+gstd.workspace = true
```
Comment (DennisInSky) [inline examples/empty/Cargo.toml:7]: Doesn't seem needed

File: examples/empty/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+sails-rtl.workspace = true
+sails-macros.workspace = true
+scale-info.workspace = true
+parity-scale-codec.workspace = true
```
Comment (DennisInSky) [inline examples/empty/Cargo.toml:11]: Doesn't seem needed

File: examples/empty/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "conditional-service"
```
Comment (DennisInSky) [inline examples/empty/Cargo.toml:2]:
```suggestion
name = "prog-without-svcs"
```
including folder name

File: examples/empty/build.rs [hunk 4]
```diff
@@ -0,0 +1,3 @@
+fn main() {
+    gwasm_builder::build();
```
[the remark this answers is not in the document]
Comment (holykol) [reply examples/empty/build.rs]: done

File: examples/empty/Cargo.toml (hunk 3 above)
Comment (holykol) [reply examples/empty/Cargo.toml]: renamed

File: examples/no-svcs-prog/app/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "no-svcs-prog-app"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+sails-macros.workspace = true
```
Comment (DennisInSky) [inline examples/no-svcs-prog/app/Cargo.toml:7]: Not required any longer. Rebase to master.

File: examples/no-svcs-prog/wasm/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "no-svcs-prog"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+no-svcs-prog-app = { path = "../app" }
+sails-macros.workspace = true
+sails-rtl.workspace = true
```
Comment (DennisInSky) [inline examples/no-svcs-prog/wasm/Cargo.toml:9]: Doubt both are needed. The first one definitely not

File: examples/no-svcs-prog/app/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#![no_std]
+
+use sails_rtl::gstd::gprogram;
+use sails_rtl::prelude::*;
```
Comment (DennisInSky) [inline examples/no-svcs-prog/app/src/lib.rs:4]: Nit: prelude doesn't seem needed either

**Review by DennisInSky — APPROVED** [review]

**Review by DennisInSky — APPROVED** [review]

**Pull request merged**